domains/client/domain-executor/src/domain_block_processor.rs

```diff
@@ -235,6 +248,22 @@ where
             on top of parent block #{header_number},{header_hash}"
         );
 
+        if let Some(to_finalize_block_number) =
+            header_number.checked_sub(&self.domain_confirmation_depth)
+        {
+            let to_finalize_block_hash =
+                self.client.hash(to_finalize_block_number)?.ok_or_else(|| {
+                    sp_blockchain::Error::Backend(format!(
+                        "Header for #{to_finalize_block_number} not found"
+                    ))
+                })?;
+            self.client
+                .finalize_block(to_finalize_block_hash, None, true)?;
```

Comment:
Do we need finality notification?

Reply:
It could be used by the tools like the explorer and such.